docs/guides/event_loop.md

```diff
@@ -2,14 +2,76 @@
 
 # An Introduction to the Event Loop in napari
 
+
+## tldr;
+
+It is not necessary to have a deep understanding of Qt or event loops to use
+napari. napari attempts to use "sane defaults" for most scenarios. Here are the
+most important details:
+
+### In IPython or Jupyter Notebook
+
+napari will detect if you are running an an IPython or Jupyter shell, and will
+automatically use the [IPython GUI event
+loop](https://ipython.readthedocs.io/en/stable/config/eventloops.html#integrating-with-gui-event-loops).
+As of [version 0.4.7](https://github.com/napari/napari/releases/tag/v0.4.7) is
+no longer necessary to call `%gui qt` manually.  Just create a viewer:
+
+```python
+In [1]: import napari
+
+In [2]: viewer = napari.Viewer()  # Viewer will show in a new window
+
+In [3]: ... # Continue interactive usage
+```
+
+````{tip}
+If you would prefer that napari did *not* start the interactive
+event loop for you in IPython, then you can disable it with:
+
+```python
+from napari.utils import SETTINGS
+
+SETTINGS.application.ipy_interactive = False
+```
+
+But then you will have to start the program yourself as described [below](#in-a-script).
+
+
+````
+### In a script
+
+Outside of IPython, you must tell napari when to "start the program" using
+{func}`napari.run`.  This will *block* execution of your script at that point,
+show the viewer, and wait for any user interaction.  When the last viewer
+closes, execution of the script will proceed.
+
+```python
+import napari
+
+viewer = napari.Viewer()
+...  # Continue setting  up your program
+
+# Start the program, show the viewer, wait for GUI interaction.
+napari.run() 
+
+# Anything below here will execute only after the viewer is closed.
+```
+
+
+-----------
+
+
```

Comment:
Can we shift the section on  "Hooking up your own events" (which includes sub-sections on keybindings, mouse events, et) up above the nitty gritty about "The Qt Event Loop".

I actually think it might also be good to move the in-depth Qt Event Loop stuff into a separate markdown file, and just link to it from here for anyone who wants more information. That would make it much clearer that these parts are "explainer" documentation on how things work under the hood. Mixing it in here with the other practical info, like how to make your own keybindings, is very likely to confuse people.

Reply:
I kinda feel like if anything should go in another file, it's the hooking up of custom events (that always felt out of place to me here).  The details about the qt event loop are the direct explanation of _why_ you need napari.run sometimes and not other times (which I think you had wanted elsewhere?)

Reply:
Yeah, these things should get split up.

Reply:
I think that's a good idea, and it's also a good idea to clearly distinguish (a) which bits of the docs people need to use/type, vs (b) the bits we are explaining so you understand the stuff under the hood better (but we don't generally want new users to try and sprinkle `QApplication` stuff through their code)